vyper/signatures/function_signature.py

```diff
@@ -78,13 +79,51 @@ def __init__(self,
         self.gas = None
         self.custom_units = custom_units
         self.nonreentrant_key = nonreentrant_key
+        self.func_ast_code = func_ast_code
+        self.calculate_arg_totals()
 
     def __str__(self):
         input_name = 'def ' + self.name + '(' + ','.join([str(arg.typ) for arg in self.args]) + ')'
         if self.output_type:
             return input_name + ' -> ' + str(self.output_type) + ':'
         return input_name + ':'
 
+    def calculate_arg_totals(self):
+        """ Calculate base arguments, and totals. """
+
+        code = self.func_ast_code
+        self.base_args = []
```

Comment:
For clarity, I think either this should go down to next to the `self.total_default_args = 0` statement, or bring that statement up here. Either way they should be together :)